src/components/header/Header.tsx

```diff
@@ -0,0 +1,38 @@
+import BurgerMenu from "./BurgerMenu";
+import "../../styles/header/header.scss";
+import Logo from "./Logo";
+import { useState } from "react";
+import SearchIcon from "../../assets/images/searchIcon.svg";
+import Cart from "../../assets/images/cart.svg";
+import Profile from "../../assets/images/profile.svg";
+
+const Header = () => {
+  const [isSearchOpen, setIsSearchOpen] = useState(false);
+
+  const toggleSearch = () => {
+    setIsSearchOpen(!isSearchOpen);
```

Comment:
it is a better practice is to do it like this: 
setIsSearchOpen(prevState => !prevState)

Reply:
ok thanks!